src/System.Windows.Forms/src/System/Windows/Forms/TextBox.cs

```diff
@@ -998,5 +983,84 @@ public TestAccessor(TextBox textBox)
 
             public bool ShouldRenderPlaceHolderText(in Message m) => _textBox.ShouldRenderPlaceHolderText(m);
         }
+
+        protected override AccessibleObject CreateAccessibilityInstance()
+           => new TextBoxAccessibleObject(this);
+
+        [ComVisible(true)]
+        internal class TextBoxAccessibleObject : ControlAccessibleObject
+        {
+            internal TextBoxAccessibleObject(TextBox owner) : base(owner)
+            {
+            }
+
+            internal new string TextLabel
+            {
+                get
+                {
+                    // Otherwise use the text of the preceding Label control, if there is one
+                    Label previousLabel = PreviousLabel;
+                    if (previousLabel != null)
+                    {
+                        string text = previousLabel.Text;
+                        if (!string.IsNullOrEmpty(text))
+                        {
+                            return text;
+                        }
+                    }
+
+                    // This control has no discernable MSAA name - return an empty string to indiciate 'nameless'.
+                    return null;
+                }
+            }
+
+            public override string Name
+            {
+                get
+                {
+                    string name = WindowsFormsUtils.TextWithoutMnemonics(TextLabel);
```

Comment:
```suggestion
                    string name = base.Name;
```
We can just use base Name, test it for null and in the case of null value provide an empty string. As I see the Accessibility proxy (this is native and explicit implementation is hidden to us) returns input text as the accessible name in case the our managed AccessibleObject returns null as AccessibleName.